linkerd/proxy/http/src/client.rs

```diff
@@ -114,48 +75,59 @@ where
     C::Future: Unpin + Send + 'static,
     C::Error: Into<Error>,
     C::Connection: Unpin + Send + 'static,
-    T: HasSettings + Clone + Send + Sync + 'static,
+    T: AsRef<Settings> + Clone + Send + Sync + 'static,
     B: hyper::body::HttpBody + Send + 'static,
     B::Data: Send,
     B::Error: Into<Error> + Send + Sync,
 {
     type Response = Client<C, T, B>;
     type Error = Error;
-    type Future = MakeFuture<C, T, B>;
+    type Future =
+        Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
 
-    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        self.connect.poll_ready(cx).map_err(Into::into)
+    fn poll_ready(&mut self, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
```

Comment:
shouldn't we still poll the `Connect` for readiness? IDK if they are ever actually not ready, but this seems like it could be a problem if the impl changed...

Reply:
We clone the connect and oneshot, so any polling we do cannot be used, anyway.

Reply:
ah, yup, i guess polling it here was never necessary. should maybe have a comment? but, not a big deal.